### PR TITLE
[SYCL] Release notes for September'20 DPCPP implementation update

### PR DESCRIPTION
**Pull request by s-kanaev**
Increase major version of libsycl.so library as well

### CODE REVIEW

**Review by pvchupin — DISMISSED** [review]

**Comment by s-kanaev** [comment]
@bader, merge if no objections

File: sycl/CMakeLists.txt [hunk 1]
```diff
@@ -12,10 +12,10 @@ option(SYCL_ADD_DEV_VERSION_POSTFIX "Adds -V postfix to version string" ON)
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
 include(AddSYCLExecutable)
 
-set(SYCL_MAJOR_VERSION 3)
+set(SYCL_MAJOR_VERSION 4)
```
Comment (bader) [inline sycl/CMakeLists.txt:15]: There are no ABI breaking changes here, so `SYCL_MAJOR_VERSION` should stay 3.
See https://github.com/intel/llvm/blob/sycl/sycl/doc/ABIPolicyGuide.md#abi-versioning-policy

**Review by bader — CHANGES_REQUESTED** [review]

File: sycl/CMakeLists.txt (hunk 1 above)
Comment (s-kanaev) [reply sycl/CMakeLists.txt]: Done. Changed to `3.1` as minor version says about non-breaking changes.

**Review by bader — APPROVED** [review]

File: sycl/CMakeLists.txt (hunk 1 above)
Comment (pvchupin) [reply sycl/CMakeLists.txt]: @bader, good catch!!

**Review by pvchupin — APPROVED** [review]

**Pull request merged**